hanadb_exporter/db_manager.py

```diff
@@ -139,3 +145,42 @@ def get_connectors(self):
         Get the connectors
         """
         return self._db_connectors
+
+    def get_active_master_host(self, hosts, master_host, master_port):
+        """
+        Check if the connection to MASTER host is up, otherwise find and return the current active MASTER host.
+        """
+        if self._system_db_connector.isconnected():
+            return master_host
+        else:
+            self._logger.info('Master host=%s is not active, scale-out is enable, looking for new Master..', str(master_host))
+            current_time = time.time()
+            timeout = current_time + SCALE_OUT_TIME_OUT
+            while current_time <= timeout:
```

Comment:
I don't like this looping timeout. I would vote of simply failing and raising an error message if the master node is not up

Reply:
We just need time till the standby node becomes master master, based on a feedback from SAP experts, it takes some minutes